analysis/quantifier-comparisons/04-cell-level-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,719 @@
+  * Number of Genes/cell 
+
+
+```{r}
+## make new data frame with mean of cells detected and standard dev
+cells_detected_stats <- cell_qc %>%
+  group_by(sample, tool) %>%
+  summarise(cells_detected = mean(cells_detected))
+cells_detected_stats
+
+## first look at cells detected
+ggplot(cells_detected_stats, aes(x = tool, y = cells_detected)) +
+  geom_jitter(mapping = aes(color = sample)) +
+  geom_violin() +
+  theme_classic() +
+  ylab("Total Cells Detected") + 
+  xlab("") + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+```
+Based on this, it looks like alevin-fry using the `--knee-distance` and kallisto do not handle single-nuclei samples well. 
+In contrast, the alevin-fry `--unfiltered-pl` samples were filtered using `DropletUtils::emptyDrops` and they have much lower numbers for the same single-nuclei samples. 
+
+Let's take a look at the mitochondrial content per cell across the tools and see if we see variation there. 
+
+
+```{r fig.height = 10, fig.width = 10}
+ggplot(cell_qc, aes(x = tool, y = subsets_mito_percent, fill = tool)) + 
+  geom_boxplot() + 
+  facet_grid(sample ~ index_type) + 
+  theme_classic() + 
+  ylab("% Mito /Cell") + 
+  theme(axis.ticks.x = element_blank(), axis.text.x = element_text(angle = 45, hjust = 1))
+```
+It looks like SCPCR000003 is a very poor sample across all tools and has a lot of dead cells. 
+It might be a good idea to remove that from our comparisons and not consider this for benchmarking. 
+Additionally, it looks like we can see that SCPCR000118 (snRNAseq) also has some variation, but the other samples tend to be fairly consistent. 
+
+
+```{r}
+# remove SCPCR000003 from cell_qc for the rest of the plotting 
+cell_qc_filter <- cell_qc %>%
+  dplyr::filter(sample != "SCPCR000003")
+```
+
+Now let's look at the number of UMI's detected/cell that we've removed our outlier sample. 
+
+```{r fig.height = 10, fig.width = 10}
+# number of UMI's/sample
+ggplot(cell_qc_filter, aes(x = tool, y = sum, fill = tool)) + 
+  geom_boxplot() + 
+  facet_grid(sample ~ index_type) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("UMI/cell") + 
+  xlab("")
+```
+At first glance, SCPCR000118 and SCPCR000119 (the single nucleus RNA-seq samples) have very low values of UMI/cell. 
+Is this because these samples are poor quality? 
+Or is it a problem with the index? 
+However, the samples don't look any better with cellranger so it could be a sample issue or a common feature of snRNA-seq samples. 
+
+Let's separate our plots to look at single cell vs. single nucleus RNA-seq.
+
+```{r fig.height=5, fig.width=10}
+# first look at single nucleus samples only 
+cell_qc_filter %>%
+  filter(seq_unit == "nucleus") %>% 
+  ggplot(aes(x = tool, y = sum, fill = tool)) + 
+  geom_boxplot() + 
+  facet_grid(sample ~ index_type) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("UMI/cell") + 
+  xlab("")
```

Comment:
Since the outliers are messing with the scale here, it might be nice to zoom in with `coord_cartesian(ylim = c(0,10000))` or something like that to see the differences here more clearly.